variantMerging.wdl

```diff
@@ -97,39 +139,70 @@ runtime {
 
 output {
   File mergedVcf = "~{outputPrefix}_mergedVcfs.vcf.gz"
+  File mergedIndex = "~{outputPrefix}_mergedVcfs.vcf.gz.tbi"
 }
 }
 
 # ==================================================================
-#  2b of 3: merge files with CombineVariants, merging matching fields
+#  3 of 3: merge files with CombineVariants, merging matching fields
 #  with priority defined by the order in the input array
 # ==================================================================
 task combineVariants {
 input {
- Array[File] inputs
+ Array[File] inputVcfs
+ Array[File] inputIndexes
  Array[String] workflows
- String reference_fasta
+ String referenceFasta
  String outputPrefix
  String modules
- Int jobMemory
- Int timeout
+ Array[String] priority = workflows
+ Int jobMemory = 12
+ Int timeout = 20
 }
 
-
 parameter_meta {
- inputTumor: "Input .bam file for analysis sample"
- inputNormal: "Optional input .bam file for control sample"
+ inputVcfs: "array of input vcf files"
+ inputIndexes: "array of tabix indexes for vcf files"
+ workflows: "array of ids of producer workflows"
+ referenceFasta: "path to the reference FASTA file"
+ outputPrefix: "prefix for output file"
+ modules: "modules for running preprocessing"
+ priority: "Comma-separated list defining priority of workflows when combining variants"
+ jobMemory: "memory allocated to preprocessing, in GB"
+ timeout: "timeout in hours"
 }
 
-# perhaps python embedded code is better here
 command <<<
-  $JAVA_HOME/bin/java -jar $GATK_ROOT/GenomeAnalysisTK.jar \
-                      -T CombineVariants \
-                      ~{sep=" " inputStrings} \ # strings "--variant:workflow FilePath" ???
-                      -R ~{reference_fasta} \
-                      -o ~{outputPrefix}_combined.vcf.gz \
-                      -genotypeMergeOptions PRIORITIZE \
-                      -priority ~{sep="," workflows}
+  python3<<CODE
+  import subprocess
+  import sys
+  inputStrings = []
+  v = "~{sep=' ' inputVcfs}"
+  vcfFiles = v.split()
+  w = "~{sep=' ' workflows}"
+  workflowIds = w.split()
+  p = "~{sep=' ' priority}"
+  priority = p.split()
+  
+  if len(vcfFiles) != len(workflowIds):
+      print("The arrays with input files and their respective workflow names are not of equal size!")
+  else:
+      for f in range(0, len(vcfFiles)):
+          inputStrings.append("--variant:" + workflowIds[f] + " " + vcfFiles[f])
+
+  javaMemory = ~{jobMemory} - 6 
+  gatkCommand  = "$JAVA_ROOT/bin/java -Xmx" + str(javaMemory) + "G -jar $GATK_ROOT/GenomeAnalysisTK.jar "
```

Comment:
"$JAVA_ROOT/bin/java" and "$GATK_ROOT/GenomeAnalysisTK.jar" probably should be parameterized as a WDL parameter.